### PR TITLE
USMON-708: Fix USM Kafka configuration

### PR DESCRIPTION
**Pull request by DanielLavie**
### What does this PR do?
Eliminated the usage of `data_streams_config.enabled` and substituted it with `service_monitoring_config.enable_kafka_monitoring`.

### Motivation
Our aim is to activate USM Kafka protocol monitoring independently of the DSM configuration value, given that this value is deprecated and no longer in use.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/config/testdata/TestDDAgentConfigYamlAndSystemProbeConfig-EnableKafka.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+service_monitoring_config:
+  enable_kafka_monitoring: true
```
Comment (guyarb) [inline pkg/network/config/testdata/TestDDAgentConfigYamlAndSystemProbeConfig-EnableKafka.yaml:2]: We don't need a real file, there is a way to have that in the test itself

Comment (guyarb) [reply pkg/network/config/testdata/TestDDAgentConfigYamlAndSystemProbeConfig-EnableKafka.yaml:2]: configurationFromYAML

**Comment by guyarb** [comment]
Also, I think this PR has to have a changelog entry

**Review by brycekahle — APPROVED** [review]
Do you need backwards compatibility?

**Review by richl-dd — APPROVED** [review]

**Comment by guyarb** [comment]
> Do you need backwards compatibility?

No, the original feature was in private beta, and we announced a while back ago we're dropping it, and no one should be using the feature and configuration

**Review by guyarb — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by hmahmood — APPROVED** [review]

**Review by git-thuerk-done — APPROVED** [review]

**Pull request merged**